### ISSUE TITLE
Error in API call: No team found with Group Id

### ISSUE THREAD

**Issue by thebenkahn** [microsoft/o365-moodle#1668]
I've started having an issue on Moodle 3.10.1, right after upgrading from version 3.10.0 to version 3.10.1 of the plugins. When the \local_o365\task\groupcreate task runs, it always fails with an error message like:

`Scheduled task failed: Create user groups in Microsoft 365 (local_o365\task\groupcreate),Error in API call: No team found with Group Id 3e2cb491-32c0-40ac-b8bb-ecb9c2138f4e`

When investigating, this ID is not in the mdl_local_o365_objects table.

In Azure, the ID is associated with a Group that was not created via the Moodle integration and does not have a Team.

When the Group is deleted in Azure, the error message still appears but with a new Group ID, also of a Group that is not associated with Moodle and does not have a related Team.

Given the above, my best guess is that we are seeing a bug related to the functionality that lets Moodle map courses to any Team in Azure that was added in the latest release: https://github.com/microsoft/o365-moodle/issues/1273 or perhaps some other new function that is trying to match non-Moodle Teams to Groups in Azure.

I am hoping we can get a resolution for this, perhaps a way to disable the Map a Course to a Team feature to see if that resolves the issue. Currently the task fails and a fail delay is added to the task, preventing regular creation/syncing of course groups.

**Comment by weilai-irl** [comment]
Hi @thebenkahn,

I have located the issue. Applying the pull requests linked to the issue should solve the problem. The fix will be included in the next release before the end of the month as well.

The error is in the function to update Teams cache, which was added to the end of the group sync tasks in the last release.

1. A Graph API function call was used to return the full list of all Teams in the organisations. This call was created according to Microsoft documentation and guideline.
2. A separate Graph API call was used on each of the Teams received that doesn't have a record in the cache table in Moodle to get the URL to the Team.

It was the second call that caused problem - some of the Teams returned in the first call were actually groups without Teams, therefore the second call failed with the error `No team found with Group Id`.

The fix to the issue is to add error handling to safely ignore groups without Teams.

Thank you for spotting and reporting the bug.

Regards,
Lai
